contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/GobblinDistcpJob.java

```diff
@@ -0,0 +1,195 @@
+package voldemort.store.readonly.mr.azkaban;
+
+import java.util.Map;
+import java.util.List;
+import java.net.URI;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.permission.FsAction;
+import org.apache.hadoop.fs.permission.FsPermission;
+import org.apache.hadoop.mapred.JobConf;
+
+import gobblin.runtime.api.JobExecutionResult;
+import gobblin.runtime.embedded.EmbeddedGobblin;
+import gobblin.runtime.embedded.EmbeddedGobblinDistcp;
+
+import voldemort.utils.Props;
+import voldemort.utils.logging.PrefixedLogger;
+import voldemort.store.readonly.swapper.AdminStoreSwapper;
+
+import azkaban.jobExecutor.AbstractJob;
+
+
+public class GobblinDistcpJob extends AbstractJob {
+    static String source;
+    private final String destination;
+    private final Props props;
+
+    GobblinDistcpJob(String id, String source, String destination, Props props) {
+        super(id, PrefixedLogger.getLogger(AdminStoreSwapper.class.getName(), destination));
+        this.source = source;
+        this.destination = destination;
+        this.props = props;
+    }
+
+    public void run() throws Exception {
+        String cdnURL = pickCDN();
+
+        if (!cdnURL.isEmpty()) {
+            info("####################################");
+            info("              Distcp");
+            info("####################################");
+            info("Using CDN: " + cdnURL);
+            try {
+                String username = props.getString("env.USER", "unknownClient");
+                String pathPrefix = props.getString(VoldemortBuildAndPushJob.PUSH_CDN_PREFIX, "/tmp/VoldemortBnP/");
+                pathPrefix = pathPrefix.endsWith("/") ? pathPrefix : pathPrefix + "/";
+                String cdnDir = source.replaceAll(".*://.*?(?=/)", cdnURL + pathPrefix + username);
+                FileSystem cdnRootFS = getRootFS(cdnDir);
+                Path from = new Path(source);
+                Path to = new Path(cdnDir);
+
+                if (!prereqSatisfied(cdnURL)) {
+                    warn("\"other_namenodes\" does not contain the CDN cluster address " + cdnURL);
+                    warn("The following steps will fail soon, and distcp will be skipped!");
+                    warn("Please add/append \"" + cdnURL + "\" to the \"other_namenodes\" attribute in your job specification.");
+                }
+
+                deleteDir(cdnRootFS, cdnDir);
+                runDistcp(from, to);
+                deleteDirOnExit(cdnRootFS, cdnDir);
```

Comment:
I don't know how useful to register a exit hook.
If the data file is very big, it could take pretty long time to delete the remote file, so will JVM still wait for the deletion?

Could we delete the data files after fetching instead of or along with registering a exit hook?
If it doesn't get deleted through the normal path, the next run will delete it.